bakery/service.go

```diff
@@ -257,9 +292,30 @@ func (svc *Service) Discharge(checker ThirdPartyChecker, id string) (*macaroon.M
 		caveats, err = checker.CheckThirdPartyCaveat(id, condition)
 	}
 	if err != nil {
+		return nil, nil, errgo.Mask(err, errgo.Any)
+	}
+	// Note that the discharge macaroon does not need to
+	// be stored persistently.
+	m, err := macaroon.New(rootKey, id, "")
+	if err != nil {
+		return nil, nil, errgo.Mask(err)
+	}
+	return m, caveats, nil
+}
+
+// Discharge calls Discharge with the service's key and uses the service
+// to add any returned caveats to the discharge macaroon.
+func (svc *Service) Discharge(checker ThirdPartyChecker, id string) (*macaroon.Macaroon, error) {
```

Comment:
I think this method still has exactly the same signature as it did before, doesn't it?
We're allowed to add functions and still keep the major version the same.